docs/content/learn/tutorials/component-definition/_index.md

```diff
@@ -0,0 +1,680 @@
+---
+component-definition:
+  ...
+  uuid: a7ba800c-a432-44cd-9075-0862cd66da6b
+  metadata:
+    title: MongoDB Component Definition Example
+    last-modified: '2001-12-17T09:30:47Z'
+    version: 20210507
+    oscal-version: 1.0.0-rc2
+    roles:
+      - id: supplier
+        title: Supplier
+    locations:
+      - uuid: 0db91129-35ed-474f-a25e-c3ff1e2cb9dc
+        address: ''
+        url: https://www.mongodb.com
+    parties:
+      - uuid: ef7c799a-c50e-49ab-83e0-515e989e6df1
+        type: prepared-for
+        name: MongoDB
+
+{{< /highlight >}}
+
+The `roles` item provides a grouping of `role` objects that each define a function assumed or expected to be assumed by a party in a specific situation.  Common examples include [component definition authors](/concepts/layer/implementation/component-definition/#component-definition-authors) (e.g., Product and Service Vendors, Capability Owners, Policy and Process Owners, Standards Bodies and Validation Bodies) and [component definition consumers](/concepts/layer/implementation/component-definition/#component-definition-consumers) (e.g., System Owners and SSP Authors).  This element has a required `id` property which expects an [NCName data type](/reference/datatypes/#ncname).  The `locations` grouping can be used to associate the metadata with any number of locations, including physical addresses and urls.  Each `location` object has an `uuid` property.  Lastly, the `parties` grouping item represents either persons or organizations that serves as the responsible entities.  There are prescribed types, including *"prepared-for"*, *"prepared-by"*, and *"approved-by"*.
+{{% /tab %}}
+{{% /tabs %}}
+
+### Component
+
+Within the component definition model, components provide a way to describe parts of an information system and the associated potential control implementations that could be established when the system is implemented.  Components are optional, but this tutorial will demonstrate their use.  In this example, we'll demonstrate how a MongoDB software component is described within an OSCAL component definition model. The snippet below provides an example of the minimum data needed to define a component.
+
+{{< tabs XML JSON YAML >}}
+{{% tab %}}
+{{< highlight xml "linenos=table" >}}
+<component-definition xmlns="http://csrc.nist.gov/ns/oscal/1.0"
+    uuid="a7ba800c-a432-44cd-9075-0862cd66da6b">
+  ...  
+  <component uuid="91f646c5-b1b6-4786-9ec3-2305a044e217" type="software">
+      <title>MongoDB</title>
+      <description>
+          <p>MongoDB is a source-available cross-platform document-oriented database program. Classified as a NoSQL database program, MongoDB uses JSON-like documents with optional schemas.</p>
+      </description>
+      <purpose>Backend NoSQL database</purpose>
+      <responsible-role role-id="supplier">
+          <party-uuid>ef7c799a-c50e-49ab-83e0-515e989e6df1</party-uuid>
+      </responsible-role>
+      <protocol name="Transmission Control" />
```

Comment:
maybe represent this as an empty `<protocol/>` element? The name here isn't really discussed until the next section. Same for the other formats.